aws/src/main/java/org/apache/iceberg/aws/s3/signer/S3V4RestSignerClient.java

```diff
@@ -136,7 +144,11 @@ private AuthSession authSession() {
       OAuthTokenResponse authResponse =
           OAuth2Util.fetchToken(httpClient(), session.headers(), credential(), SCOPE);
       return AuthSession.fromTokenResponse(
-          httpClient(), TOKEN_REFRESH_EXECUTOR, authResponse, startTimeMillis, session);
+          httpClient(),
+          keepTokenRefreshed() ? TOKEN_REFRESH_EXECUTOR : null,
```

Comment:
Right now `TOKEN_REFRESH_EXECUTOR` is a static final here. I'll open a separate PR so that it doesn't get initialized when token refresh is disabled

Reply:
here's the PR I mentioned: https://github.com/apache/iceberg/pull/7046